src/server/pps/server/api_server.go

```diff
@@ -1954,8 +1967,18 @@ func RepoNameToEnvString(repoName string) string {
 	return strings.ToUpper(repoName)
 }
 
-// Convert a persist.JobInfo into a Kubernetes batch.Job spec
-func job(kubeClient *kube.Client, jobInfo *persist.JobInfo, jobShimImage string, jobImagePullPolicy string) (*batch.Job, error) {
+type jobOptions struct {
+	labels             map[string]string
+	parallelism        int32
+	userImage          string
+	jobShimImage       string
+	jobImagePullPolicy string
+	jobEnv             []api.EnvVar
+	volumes            []api.Volume
+	volumeMounts       []api.VolumeMount
+}
+
+func getJobOptions(kubeClient *kube.Client, jobInfo *persist.JobInfo, jobShimImage string, jobImagePullPolicy string) (*jobOptions, error) {
```

Comment:
Small nit, would you be opposed to just calling this `jobOptions`? I think "get" in function names is often superfluous, if you're reading code and see `jobOptions(...)` you generally assume that evaluating that expression is going to get you `jobOptions`. Also this will make it match the functions around it which are `job` and `service` not `getJob` and `getService`.